src/Sarif.Viewer.VisualStudio/ErrorList/SarifTableDataSource.cs

```diff
@@ -98,117 +93,191 @@ public string Identifier
 
         public string DisplayName
         {
-            get { return Constants.VSIX_NAME; }
+            get { return Resources.ErrorListTableDataSourceDisplayName; }
         }
 
         public IDisposable Subscribe(ITableDataSink sink)
         {
-            return new SinkManager(this, sink);
+            var sarifTableDataSourceSink = new SarifTableDataSourceSink(sink);
+            sarifTableDataSourceSink.Disposed += this.TableSink_Disposed;
+
+            using (this.sinksLock.EnterWriteLock())
+            {
+                sinks.Add(sarifTableDataSourceSink);
+            }
+
+            IImmutableList<SarifResultTableEntry> entriesToNotify;
+
+            using (this.tableEntriesLock.EnterReadLock())
+            {
+                entriesToNotify = logFileToTableEntries.Values.SelectMany((tableEntries) => tableEntries).ToImmutableList();
+            }
+
+            sarifTableDataSourceSink.AddEntries(entriesToNotify);
+
+            return sarifTableDataSourceSink;
         }
         #endregion
 
-        public void AddSinkManager(SinkManager manager)
+        public void AddErrors(IEnumerable<SarifErrorListItem> errors)
         {
-            // This call can, in theory, happen from any thread so be appropriately thread safe.
-            // In practice, it will probably be called only once from the UI thread (by the error list tool window).
-            lock (_managers)
+            if (errors == null)
             {
-                _managers.Add(manager);
+                return;
             }
-        }
 
-        public void RemoveSinkManager(SinkManager manager)
-        {
-            // This call can, in theory, happen from any thread so be appropriately thread safe.
-            // In practice, it will probably be called only once from the UI thread (by the error list tool window).
-            lock (_managers)
+            var tableEntries = errors.Select((error) => new SarifResultTableEntry(error)).ToImmutableList();
+
+            this.CallSinks((sink) =>
```

Comment:
>) [](start = 32, length = 1)

Tiniest of nits: you don't need the parens when the lambda has only one argument.